### PR TITLE
Issue #29880, issue #28480: Document/Image mass uploader feature

### PR DESCRIPTION
**Pull request by andersdd**
requires xtuple/xtuple#3146

Also now supports UPDATE and APPEND actions

### CODE REVIEW

File: plugin/csvtoolwindow.cpp [hunk 1]
```diff
@@ -465,6 +470,8 @@ bool CSVToolWindow::importStart()
   QString value;
   QString label;
   QVariant var;
+  QString  _fileName;
```
Comment (gilmoskowitz) [inline plugin/csvtoolwindow.cpp]: The `_` prefix looks wrong on `_fileName`.

File: plugin/csvtoolwindow.cpp [hunk 2]
```diff
@@ -536,6 +544,46 @@ bool CSVToolWindow::importStart()
...
+                {
+                  var = docLoadAndEncode(_fileName);
+                  if (var == false)
+                    var = QVariant(QString::null); // Nothing (error)
+                  break;
+                }
+                default:
+                  var = QVariant(value);
+              }
+            }
```
Comment (gilmoskowitz) [inline plugin/csvtoolwindow.cpp]: Looks like the `if (var == false)` checks could all be moved after the `switch`; `value` should never be `false` here.

File: plugin/csvmap.cpp [hunk 3]
```diff
@@ -520,5 +567,44 @@ QStringList CSVMapField::actionList()
   list << "UseEmptyString";
   list << "UseAlternateValue";
   list << "UseNull";
+  list << "SetColumnFromDataFile";
+  return list;
+}
+
+QString CSVMapField::fileTypeToName(FileType ft)
+{
+  QString str = "N/A";
```
[the remark this answers is not in the document]
Comment (andersdd) [reply plugin/csvmap.cpp]: @gilmoskowitz  Understand the concept and usage of QMap but struggling with static functions in this area causing flow on issues/errors.

**Comment by gilmoskowitz** [comment]
+1 when https://github.com/xtuple/xtuple/pull/3146 ci failure is fixed

**Pull request merged**